### PR TITLE
Adds e2e's for PGL pages

### LINKED ISSUE #5154: Create E2E's for PGL pages
**Is your feature request related to a problem? Please describe.**
Now that work has begun to get CPS PGL assets to render with Simorgh we need to create basic e2e tests which will check if a PGL asset exists.

**Describe the solution you'd like**
- [x] Create a new directory in our page e2e's for PGL pages, with test files that follow a similar structure to our MAP pages [tests](https://github.com/bbc/simorgh/tree/latest/cypress/integration/pages/mediaAssetPage). 
- [x] Also, ensure that the PGL page type is being tested for 404's and other relevant error codes. 
- [x] We'll need to check that are some content items on the PGL asset e.g headline, summary, images etc.

Here is a local asset page: http://localhost:7080/pidgin/sport-23252855.

~For now, this is the only local asset so the test should be omitted for other services that aren't Pidgin.~

Local PGL fixtures have now been added for all services through this PR: https://github.com/bbc/simorgh/pull/5168

**Describe alternatives you've considered**

**Testing notes**
Check with a tester to see if e2e's are written correctly.

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### LINKED ISSUE #5158: Add description, OG and Twitter metadata to PGLs
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**
Ensure PGLs contain the following metadata.

The metadata logic can be found in the metadata [container](https://github.com/bbc/simorgh/blob/latest/src/app/containers/Metadata/index.jsx). And the container is used in the [CpsAssetPageMain](https://github.com/bbc/simorgh/blob/latest/src/app/containers/CpsAssetPageMain/index.jsx#L57) to produce the metadata for that asset.

We need to ensure that other page types like MAP pages can make use of the metadata that is being added for PGL pages and if not we limit it per page asset type. Speak to the Media team to check this.

Generic:
- [x] description = Promo summary - Check with the Media team to see if the current is okay https://github.com/bbc/simorgh/blob/887963dd6cdc1242f43ca0eab5a1167f3f04d11e/src/app/containers/CpsAssetPageMain/index.jsx#L28

Open Graph (Facebook):
- [x] og:title =	Promo Headline Headline
- [x] og:type =	article	[unchanged from article data]
- [x] og:description =	Promo summary 
- [x] og:site_name =	Service name	[unchanged from article data]
- [x] og:locale =	Service locale	[unchanged from article data]
- [x] og:url =	Canonical URL of article	[unchanged from article data]
- [x] og:image = Promo -> IndexImage (same as promo image used on FP ).  If it doesn't exist, use Service -> Logo
- [x] og:image:alt = Promo -> IndexImage -> AltText. If it doesn't exist, use the Service -> Logo -> AltText
- [x] article:author = should be article author [from service config](https://github.com/bbc/simorgh/search?p=2&q=articleAuthor%3A+%27&unscoped_q=articleAuthor%3A+%27) e.g `https://www.facebook.com/bbcnews`

Twitter:
- [x] twitter:creator = Service name e.g. `@BBCMundo` [unchanged from article data]
- [x] twitter:site = Service name e.g. `@BBCMundo` [unchanged from article data]
- [x] twitter:title =	Promo Headline Headline
- [x] twitter:description =	Promo Summary
- [x] twitter:img:src = Promo -> IndexImage (same as promo image used on FP ).  If it doesn't exist, use Service -> Logo
- [x] twitter:image:alt = Promo -> IndexImage -> AltText. If it doesn't exist, use the Service -> Logo -> AltText

**Describe alternatives you've considered**

**Testing notes**
Check if all other metadata that is present is the same as our current articles pages. -- Have checked this and added the metadata that is present on the live article in comments below.

https://developers.facebook.com/tools/debug/ -  Validator for Facebook metadata
https://cards-dev.twitter.com/validator - Validator for Twitter metadata
These tools might only work with live URL's and not with code snippets. So more investigation might need to be done to find other methods of validation of these two instances of metadata.

Existing snapshots will need updating or adding when the correct metadata has been added to the asset data.

Check if there any existing and relevant e2e's and if they need to be updated/removed.

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

**Additional context**
- A reference to similar work: https://github.com/bbc/simorgh/pull/5075

### PR DESCRIPTION
**Pull request by HarveyPeachey**
Resolves https://github.com/bbc/simorgh/issues/5154, and https://github.com/bbc/simorgh/issues/5158 by adding PGL's to metadata logic.

**Overall change:** 
Created basic e2e's for the new PGL (Photo Gallery) asset page types added into Simorgh.

**Code changes:**
- Added all of our local PGL asset locations into our service config file.
- Add tests:
  - that checks `h1` content on the page matches asset's `body.promo.headlines.headline` .
  - that MAP pages use to check for a timestamp on the page.
  - that checks the description/summary of the page.
  - for canonical that checks if an `img` is visible.
  - for amp that checks if an `amp-img` is visible.
- Add PGL type to imageAltText and imageSrc check.
- Add PGL type to 'shared metadata' e2e test.

**Notes for tester 👩‍💻:**
- This PR also resolves https://github.com/bbc/simorgh/issues/5158, so we will need to verify that the requirements of this issue are met. I.E. All metadata tags are being output as expected.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [x] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [x] This PR requires manual testing

### CODE REVIEW

**Comment by HarveyPeachey** [comment]
~Blocked until https://github.com/bbc/simorgh/issues/5230 is resolved as ATI e2e tests ran on all pages isn't working for PGL pages.~

Unblocked as the issue has been resolved

File: cypress/integration/pages/photoGalleryPage/tests.js
Comment (andrewscfc) [inline cypress/integration/pages/photoGalleryPage/tests.js]: Sorry pretty new to Cypress, which element does this retrieve, I found these two `time` elements on the PGL page:
image

File: cypress/integration/pages/photoGalleryPage/tests.js
Comment (andrewscfc) [inline cypress/integration/pages/photoGalleryPage/tests.js]: Look at the docs for `get` not really seeing an a situation where it would return `0`: https://docs.cypress.io/api/commands/get.html#Syntax

File: cypress/integration/pages/photoGalleryPage/tests.js
Comment (andrewscfc) [inline cypress/integration/pages/photoGalleryPage/tests.js]: What is the significance of the single minute tolerance? Is this because some test fixtue data is unsuitable?

File: cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js
Comment (andrewscfc) [inline cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js]: Do we want to keep this commented out code?

File: cypress/integration/pages/photoGalleryPage/testsForCanonicalOnly.js
Comment (andrewscfc) [inline cypress/integration/pages/photoGalleryPage/testsForCanonicalOnly.js]: Commented out code

**Review by andrewscfc — COMMENTED** [review]
Most of my comments will probably just help me get familiar with our cypress tests, thanks in advance for you time!

File: cypress/integration/pages/photoGalleryPage/tests.js
Comment (HarveyPeachey) [reply cypress/integration/pages/photoGalleryPage/tests.js]: So this `get('time)` retrieves all the time elements and the `.eq(0)` grabs the first `<time>` tag. I'm also trying to understand cypress as I haven't written many tests before this :).

File: cypress/integration/pages/photoGalleryPage/tests.js
Comment (HarveyPeachey) [reply cypress/integration/pages/photoGalleryPage/tests.js]: See above comment about `.eq(0)`.

File: cypress/integration/pages/photoGalleryPage/tests.js
Comment (HarveyPeachey) [reply cypress/integration/pages/photoGalleryPage/tests.js]: Yep, there’s a bug with some CPS assets where `firstPublished` and `lastPublished` are milliseconds apart when they've only been published once. So the media team created logic to give tolerance for this discrepancy. They wrote a test for MAP pages which took this into account so I re-used it for PGL assets as we're using the same logic.

File: cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js
Comment (HarveyPeachey) [reply cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js]: We can get rid of it, I just commented it out just in case we needed it for writing future tests. Not sure what the standard procedure is for that but happy to remove them.

File: cypress/integration/pages/photoGalleryPage/tests.js
Comment (simonsinclair) [inline cypress/integration/pages/photoGalleryPage/tests.js]: We could prune `.should('exist')` from this chain as `cy.get('time')` followed by `.should('be.visible')` will verify this element exists.

File: cypress/integration/pages/photoGalleryPage/tests.js
Comment (simonsinclair) [inline cypress/integration/pages/photoGalleryPage/tests.js]: In the test above, we verify the element's visible. Should this one be visible too?

File: cypress/integration/pages/photoGalleryPage/tests.js
Comment (simonsinclair) [inline cypress/integration/pages/photoGalleryPage/tests.js:25]: This is much more readable than `.eq(0)`. 👍

File: cypress/integration/pages/photoGalleryPage/tests.js
Comment (simonsinclair) [inline cypress/integration/pages/photoGalleryPage/tests.js]: Would it make sense to improve the readability of this condition? E.G.
```js
const hasTimestampPrefix = timeDifferenceMinutes > minutesTolerance;
// Assertions.
if (hasTimestampPrefix) {
  // Additional assertions.
}
```

File: cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js
Comment (simonsinclair) [reply cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js]: I would support removing these until they're needed. 👍

File: cypress/integration/pages/photoGalleryPage/testsForCanonicalOnly.js
Comment (simonsinclair) [inline cypress/integration/pages/photoGalleryPage/testsForCanonicalOnly.js]: Would it make sense to be more economical with our selection if we're just planning to verify that there is "at least one" visible image on the page? E.G.
```js
cy.get('img')
  .first()
  .should('be.visible');
```

File: cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js
Comment (simonsinclair) [inline cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js]: Please see my comment on the Canonical version of this test. 😊

File: cypress/integration/pages/photoGalleryPage/tests.js
Comment (andrewscfc) [reply cypress/integration/pages/photoGalleryPage/tests.js]: Oh, so `eq` gets by index: https://docs.cypress.io/api/commands/eq.html#Syntax !

That makes sense now, I assumed it was some kind of assertion, i.e. equals 0. I'm still unsure why the api is `eq`!

File: cypress/integration/pages/photoGalleryPage/tests.js
Comment (andrewscfc) [reply cypress/integration/pages/photoGalleryPage/tests.js]: Thanks makes sense now

File: cypress/integration/pages/photoGalleryPage/tests.js
Comment (andrewscfc) [reply cypress/integration/pages/photoGalleryPage/tests.js]: You may want to add a comment in both the MAP and PGL tests just to make that clear, seems pretty arbitrary without the context.

File: cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js
Comment (andrewscfc) [reply cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js]: If we have a convention of leaving it around for future tests no worries, just checking it wasn't accidental.

File: cypress/integration/pages/photoGalleryPage/tests.js
Comment (simonsinclair) [inline cypress/integration/pages/photoGalleryPage/tests.js]: I'm not sure understand the title of this test. 😅Would it make sense to prune the bit about displaying styled text?

File: cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js
Comment (simonsinclair) [inline cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js:16]: Hmm I can see we're following a pattern here, but I'm curious why we don't do this in testsForAllPages. Do you happen to know why?

**Review by simonsinclair — COMMENTED** [review]
Super work, Harvey. I left a couple of comments below.

File: cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js
Comment (HarveyPeachey) [reply cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js:16]: I asked @sareh about this specific test because I wondered the same thing and this test should be placed in `testsForAllAmpPages`. But It'll have to be excluded for 404 error pages and article page. 

Wouldn't suppose you would know why this test isn't ran on article pages?

Comment (simonsinclair) [reply cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js:16]: Having done a bit of digging, I found this on line 22 of runTestsForPage. Each page visited by Cypress using this function is tested for a 200. 

```js
const visitPage = (path, pageType) => {
  const expectedContentType = 'text/html';
  const isErrorPage = pageType.includes('error');
  const expectedStatus = isErrorPage ? 404 : 200;
  const failOnStatusCode = !isErrorPage;

  cy.testResponseCodeAndType(path, expectedStatus, expectedContentType);
  cy.visit(path, {
    failOnStatusCode,
  });
};
```

This needs verifying, but we appear to be manually duplicating this test in the following files:
- frontPage AMP only
- liveRadio AMP only
- mediaAsset AMP only
- _photoGallery AMP only_

File: cypress/integration/pages/photoGalleryPage/tests.js
Comment (HarveyPeachey) [reply cypress/integration/pages/photoGalleryPage/tests.js]: I agree with all of your suggestions for this test. Probably need a review from someone in the Media team as I just used their test for MAP pages.

File: cypress/integration/pages/photoGalleryPage/tests.js
Comment (HarveyPeachey) [reply cypress/integration/pages/photoGalleryPage/tests.js]: Yep good idea. I also did a bit of digging into cypress docs and learned about this https://docs.cypress.io/api/commands/should.html#Effect-on-default-DOM-assertions.

File: cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js
Comment (andrewscfc) [reply cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js:16]: Could we resolve this duplication in a follow-up PR?

File: cypress/integration/pages/testsForAllPages.js
Comment (andrewscfc) [inline cypress/integration/pages/testsForAllPages.js:443]: Did you decide against commenting to explain the need for this logic? Happy to merge if we think it's better not to comment

**Review by andrewscfc — COMMENTED** [review]
Just a question over commenting, otherwise happy to approve 🙂

File: cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js
Comment (simonsinclair) [inline cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js]:
```suggestion
      cy.get('figure')
        .first()
        .find('amp-img')
        .should('be.visible');
```

Comment (simonsinclair) [reply cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js]: Because the `amp-img` we care about is nested within our first selection, we can continue to use that by chaining from it. 🙂

File: cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js
Comment (simonsinclair) [reply cypress/integration/pages/photoGalleryPage/testsForAMPOnly.js:16]: @andrewscfc yes, I think so. I've raised an issue here: https://github.com/bbc/simorgh/issues/5284.

File: cypress/integration/pages/photoGalleryPage/testsForCanonicalOnly.js
Comment (simonsinclair) [inline cypress/integration/pages/photoGalleryPage/testsForCanonicalOnly.js]: The suggested optimisation on the AMP equivalent could also be made here.

File: cypress/integration/pages/testsForAllPages.js
Comment (simonsinclair) [inline cypress/integration/pages/testsForAllPages.js:224]: This doesn't need to be tackled here, but (for visibility) I've raised an issue here https://github.com/bbc/simorgh/issues/5280 to refactor this switch to make use of our page type constants.

**Review by simonsinclair — APPROVED** [review]
Great work in migrating the timestamp tests to ForAllPages.
I've left a suggestion, but otherwise this PR looks good. 👍

File: cypress/integration/pages/testsForAllPages.js
Comment (HarveyPeachey) [reply cypress/integration/pages/testsForAllPages.js:443]: Whoops, must've forgotten to copy it over when I merged the tests. I'll add it now.

Comment (andrewscfc) [reply cypress/integration/pages/testsForAllPages.js:443]: Thanks @HarveyPeachey !

**Review by andrewscfc — APPROVED** [review]

**Review by mykeels — APPROVED** [review]

**Comment by paruchurisilpa** [comment]
checked using the link:
URL --> http://localhost:7080/pidgin/sport-23252855

name="og:title" content="PGL - Blood pressure drugs dey work better - BBC News Pidgin"/>
name="og:type" content="article"/>
name="og:description" content="This is a test asset used in Simorgh, please do not edit it without asking."/>
property="og:site_name" content="BBC News Pidgin"/
name="og:locale" content="pcm"/>
name="og:url" content="http://localhost:7080/pidgin/sport-23252855"/>
name="og:image" content="https://news.files.bbci.co.uk/ws/img/logos/og/pidgin.png"/>
name="og:image:alt" content="BBC News Pidgin"/>
name="article:author" content="https://www.facebook.com/bbcnews"/>

Twitter:
name="twitter:creator" content="@BBCNews"/>
name="twitter:site" content="@BBCNews"/>
name="twitter:title" content="PGL - Blood pressure drugs dey work better - BBC News Pidgin"/>
 twitter:img:src —> Not available
twitter:image:alt —> Not available

**Comment by HarveyPeachey** [comment]
> checked using the link:
> URL --> http://localhost:7080/pidgin/sport-23252855
> 
> name="og:title" content="PGL - Blood pressure drugs dey work better - BBC News Pidgin"/>
> name="og:type" content="article"/>
> name="og:description" content="This is a test asset used in Simorgh, please do not edit it without asking."/>
> property="og:site_name" content="BBC News Pidgin"/
> name="og:locale" content="pcm"/>
> name="og:url" content="[http://localhost:7080/pidgin/sport-23252855"/>](http://localhost:7080/pidgin/sport-23252855%22/%3E)
> name="og:image" content="[https://news.files.bbci.co.uk/ws/img/logos/og/pidgin.png"/>](https://news.files.bbci.co.uk/ws/img/logos/og/pidgin.png%22/%3E)
> name="og:image:alt" content="BBC News Pidgin"/>
> name="article:author" content="[https://www.facebook.com/bbcnews"/>](https://www.facebook.com/bbcnews%22/%3E)
> 
> Twitter:
> name="twitter:creator" content="@BBCNews"/>
> name="twitter:site" content="@BBCNews"/>
> name="twitter:title" content="PGL - Blood pressure drugs dey work better - BBC News Pidgin"/>
> twitter:img:src —> Not available
> twitter:image:alt —> Not available

@paruchurisilpa Weird I just tested it out as well and I'm not seeing any errors about missing tags? We can address this in a separate issue.

Screenshot 2020-02-03 at 16 51 37

**Pull request merged**
